### PR TITLE
Added wall sliding and jumping, also fixed a bug where jumping was not being set to false

### PR DESCRIPTION
**Pull request by paul-mcnamee**
These changes mainly leverage the is_on_wall_only bool from CharacterBody2D. You can disable/enable the feature via the exported bool ENABLE_WALL_JUMPING.

Here is a video recording of the wall jumping / sliding behavior:
https://github.com/dragon1freak/PlatformerCharacterController/assets/35277807/367042a4-2a7b-4c15-b26b-23853bde065f

### CODE REVIEW

**Comment by paul-mcnamee** [comment]
Here is a recording of the wall jumping / sliding behavior:
https://github.com/dragon1freak/PlatformerCharacterController/assets/35277807/367042a4-2a7b-4c15-b26b-23853bde065f

File: src/scripts/PlatformerController.gd [hunk 1]
```diff
@@ -89,11 +97,14 @@ func manage_state() -> void:
 		else:
 			state = WALK
 	elif velocity.y < 0:
-		state = JUMP
+			state = JUMP
```
Comment (dragon1freak) [inline src/scripts/PlatformerController.gd]: This should work as is but we should keep the indentation pattern the same
```suggestion
		state = JUMP
```

**Comment by dragon1freak** [comment]
Other than that indentation difference this all looks good!  Very similar to how I implemented it in a jam game, I'll double check it works fine on my end and merge it in.

**Comment by dragon1freak** [comment]
Works great, nice job!

**Pull request merged**